.github/workflows/deploy.yaml

```diff
@@ -8,20 +8,138 @@ on:
+      run: npm install -g yarn
+
+    - name: Install Poetry
+      uses: Gr1N/setup-poetry@v9
+      with:
+        poetry-version: "1.8.0"
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+        cache: poetry
+        cache-dependency-path: poetry.lock
+
+    - name: Assume Github OIDC role
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-to-assume: ${{ vars.DEPLOY_ROLE }}
+        role-session-name: federated-collection-discovery-${{ github.event.inputs.deployment_environment }}-deploy
+    
+    - name: Install deployment dependencies
+      run: | 
+        npm install
+        poetry install
+        cd client_app/src/client
+        yarn install
+
+    - name: Run CDK deploy
+      env:
+        STAGE: "${{ github.event.inputs.deployment_environment }}"
+        API_VERSION: "${{ github.event.inputs.api_version }}"
+        API_CERTIFICATE_ARN: "${{ vars.API_CERTIFICATE_ARN }}"
+        API_DOMAIN_NAME: "${{ vars.API_DOMAIN_NAME }}"
+        CLIENT_CERTIFICATE_ARN: "${{ vars.CLIENT_CERTIFICATE_ARN }}"
+        CLIENT_DOMAIN_NAME: "${{ vars.CLIENT_DOMAIN_NAME }}"
+        STAC_API_URLS: ${{ vars.STAC_API_URLS }}
+      run: |
+        start_time=$(date +%s)
+        poetry run npx cdk deploy --all --require-approval never --outputs-file cdk-output.json
+        status=$?
+        end_time=$(date +%s)
+        duration=$((end_time - start_time))
+        echo "success=${status}" >> $GITHUB_ENV
+        echo "duration=${duration}" >> $GITHUB_ENV
+        if [ $status -eq 0 ]; then
+          echo "CDK deploy successful."
+        else
+          echo "CDK deploy failed." >&2
+          exit 1
+        fi
+        export STACK_NAME=$(jq -r 'keys_unsorted[0]' cdk-output.json)
+        export REACT_APP_API_URL=$(jq -r --arg stack_name "$STACK_NAME" '.[$stack_name].ApiEndpoint' cdk-output.json)
+        echo "REACT_APP_API_URL=$REACT_APP_API_URL" >> $GITHUB_ENV
+        export BUCKET_NAME=$(jq -r --arg stack_name "$STACK_NAME" '.[$stack_name].ClientBucketName' cdk-output.json)
+        echo "BUCKET_NAME=$BUCKET_NAME" >> $GITHUB_ENV
+        export DISTRIBUTION_ID=$(jq -r --arg stack_name "$STACK_NAME" '.[$stack_name].CloudFrontId' cdk-output.json)
+        echo "DISTRIBUTION_ID=$DISTRIBUTION_ID" >> $GITHUB_ENV
+
+        echo "duration=${duration}" >> "$GITHUB_OUTPUT"
+        echo "api_endpoint=${REACT_APP_API_URL}" >> "$GITHUB_OUTPUT"
+        echo "client_file_bucket=${BUCKET_NAME}" >> "$GITHUB_OUTPUT"
+        echo "cloudfront_distribution=${DISTRIBUTION_ID}" >> "$GITHUB_OUTPUT"
+
+    - name: Build and deploy client app
+      working-directory: client_app/src/client
+      run: | 
+        yarn build
+        aws s3 sync ./build s3://${BUCKET_NAME}
+        # invalidate existing index.html to refresh the site
+        aws cloudfront create-invalidation --distribution-id ${DISTRIBUTION_ID} --paths "/index.html"
```

Comment:
I used to create invalidations for assets/js as well, but recently I've just been doing wildcard invalidations - I think it's a nice to have, especially when things are unexpectedly cached